docs/static/examples/development.yaml

```diff
@@ -94,41 +94,165 @@ write_files:
-        helm repo add united-manufacturing-hub https://repo.umh.app/
-        echo "Added helm repo!"
-        echo "Creating namespaces factorycube-edge and factorycube-server..."
-        kubectl create namespace factorycube-edge && kubectl create namespace factorycube-server
-        echo "Created namespaces!"
-        echo "Install factorycube-edge..."
-        helm install factorycube-edge united-manufacturing-hub/factorycube-edge --values "/home/rancher/configs/factorycube-edge-development.yaml" --set serialNumber=$(hostname) --kubeconfig /etc/rancher/k3s/k3s.yaml -n factorycube-edge --devel
-        echo "Installed factorycube-edge!"
-        echo "Install factorycube-server..."
-        helm install factorycube-server united-manufacturing-hub/factorycube-server --kubeconfig /etc/rancher/k3s/k3s.yaml -n factorycube-server --devel
-        echo "Installed factorycube-server!"
-
-        echo "Enabling SSH password authentification..."
-        sudo vim /etc/ssh/sshd_config -c "%s/PasswordAuthentication  no/PasswordAuthentication  yes/g | write | quit" && sudo service sshd restart
-        echo "Enabled SSH password authentification!"
-        echo "Initial setup finished! This might take now awhile..."
-    else
-        echo "Already configured. If you want to reconfigure delete /home/rancher/DO_NOT_DELETE"
-    fi
+      #!/bin/bash
+      LOCKDIR=/home/rancher/DO_NOT_DELETE
+
+      # This runs as root
+
+      if [ -d $LOCKDIR ]; then
+      echo "Already configured. If you want to reconfigure delete /home/rancher/DO_NOT_DELETE"
+      exit
+      fi
+
+
+      # Restart as rancher
+      if [ $UID -eq 0 ]; then
+      echo "Initial setup! Waiting 20 seconds for full startup..."
+      #sleep 20
+
+
+      ###
+      # Check if NTP server is reachable
+      ###
+
+      # Get both ntp servers from config
+      NTP1=$(cat /k3os/system/config.yaml | grep ntp_servers -A2 | grep - | sed 's/  - //g' | sed -n 1p)
+      NTP2=$(cat /k3os/system/config.yaml | grep ntp_servers -A2 | grep - | sed 's/  - //g' | sed -n 2p)
+
+      echo "Testing NTP server $NTP1"
+      echo "Testing NTP server $NTP2"
+
+      # Try 10 times to reach either of them
+      i="0"
+      reachable_ntp_server_found=0
+      while [ $i -lt 10 ]
+      do
+      if ! ping -c1 -W1 $NTP1 > /dev/null 2>&1 ; then
+      echo "$NTP1 not reachable_ntp_server_found !"
+      sleep 10
+      else
+      echo "$NTP1 is reachable_ntp_server_found !"
+      reachable_ntp_server_found=1
+      fi
+
+      if ! ping -c1 -W1 $NTP2 > /dev/null 2>&1 ; then
+      echo "$NTP2 not reachable_ntp_server_found !"
+      sleep 10
+      else
+      echo "$NTP2 is reachable_ntp_server_found !"
+      reachable_ntp_server_found=1
+      fi
+      if [ $reachable_ntp_server_found -eq 1 ]; then
+      break
+      fi
+      i=$[$i+1]
+      done
+
+      if [ $reachable_ntp_server_found -eq 0 ]; then
+      echo "No reachable NTP server found, aborting setup"
```

Comment:
Please add a note for someone who does not know what a NTP server is, e.g., "No reachable NTP server found, aborting setup. This can happen when you either have no connection to the internet or when you are in a limited network that blocks NTP requests (some universities do that). The solution is to connect ot the internet and / or to specify the allowed NTP servers manually in the development.yaml. Fore more information visit docs.umh.app"

Reply:
also the script should not abort when the time is configured properly (e.g., in BIOS). For this you can place the additional time check as a "else if" to this "if" clause